docs/sandboxing/README.md

```diff
@@ -10,9 +10,95 @@ There is no clear solution to this yet, but it should take heavy inspiration fro
 
 ### Linux
 
-We could use something like flatpak here, though its unlikely, the current most likely solution is to use linux namespaces to work like a docker-like system.
-
-This is mainly solved other then implementation details
+Linux offers different kinds of isolation using namespacing.
+A fairly low-level tool to access these capabilities is `unshare` that allows creating new namespaces for child processes.
+`unshare` is fine, if full-isolation should be achieved, but we definitely need *some* access to the host system, and without integrating with multiple tools, `unshare` by itself is undesirable.
+A popular tool that utilizes the same capabilities as `unshare` is `bubblewrap`, which is most prominently used by Flatpak. It also allows creating new namespaces, just like `unshare` but additionally provides easy ways to bind paths from the host into the mount namespace.
+
+#### Bubblewrap
+
+Using `bubblewrap` is quite simple. It can be configured by running the `bwrap` binary and defining command line arguments.
+As the name suggests it acts as a wrapper, so after specifying bwrap options the main program inside the sandbox can be specified.
+
+##### Base configuration
+
+- `--unshare-all --share-net` - Try to create all kinds of namespaces except for networking
+  - Doing unprivileged networking in userspace is slow, which is why we will NOT sandbox the root networking namespace
+- `--die-with-parent` - Kill sandbox if launcher crashes/exits
+- `--dev /dev --dev-bind-try /dev/dri /dev/dri` - Allow access to DRM devices for hardware accelerated rendering
+- `--proc /proc` - Mount namespaced procfs at /proc
```

Comment:
I don't think it's safe to expose entire procfs for minecraft?

Reply:
If I understand the proposal correctly, a procfs scoped to the new pid namespace would be mounted. This only shows the processes within the sandbox, which should be safe.